docs/proposals/04-etcd-member-custom-resource.md

```diff
@@ -0,0 +1,561 @@
+
+In the following sub-sections, the state transitions are categorized into several flows making it easier to grasp the different transitions.
+
+#### Top Level State Transitions
+
+Following DFA represents top level state transitions (without any representation of sub-states). As described in the table above there are 4 top level states:
+
+* `New`- this is a start state for all newly created etcd-members
+
+* `Initializing` - In this state backup-restore will perform pre-requisite actions before it triggers the start of an etcd process. DB validation and optionally restoration is done in this state. Possible sub-states are: `DBValidationSanity`, `DBValidationFull` and `Restoration`
+
+* `Starting` - Once the optional initialization is done backup-restore will trigger the start of an etcd process. It can either directly go to `Learner` sub-state or wait for getting added as a learner and therefore be in `PendingLearner` sub-state.
+
+* `Started` - In this state the etcd-member is a full voting member. It can either be in `Leader` or `Follower` sub-states.
+
+<img title="" src="assets/04-top-level-state-transitions.excalidraw.png" alt="" data-align="center">
+
+#### Starting an Etcd-Member in a Single-Node Etcd Cluster
+
+Following DFA represents the states, sub-states and transitions of a single etcd-member for a cluster that is bootstrapped from cluster size of 0 -> 1.
+
+<img title="" src="assets/04-state-transitions-bootstrap-0-1.excalidraw.png" alt="" data-align="center">
+
+#### Addition of a New Etcd-Member in a Multi-Node Etcd Cluster
+
+Following DFA represents the states, sub-states and transitions of an etcd cluster which starts with having a single member (Leader) and then one or more new members are added which represents a scale-up of an etcd cluster from 1 -> n, where n is [odd](https://etcd.io/docs/v3.5/faq/#why-an-odd-number-of-cluster-members).
+
+<img title="" src="assets/04-state-transitions-scaleup-1-n.excalidraw.png" alt="" data-align="center">
+
+#### Restart of a Voting Etcd-Member in a Multi-Node Etcd Cluster
+
+Following DFA represents the states, sub-states and transitions when a voting etcd-member in a multi-node etcd cluster restarts.
+
+<img title="" src="assets/04-state-transitions-restart-member.excalidraw.png" alt="" data-align="center">
+
+> NOTE: If the DB validation fails then data directory of the etcd-member is removed and etcd-member is removed from cluster membership, thus transitioning it to `New` state. The state transitions from `New` state are depicted by [this section](#addition-of-a-new-etcd-member-in-a-multi-node-etcd-cluster).
```

Comment:
We are not doing anything different for failure of DVF-S and DVF-F. In both the cases, we are just restarting.

Reply:
Yes. Is this an observation or is there a question?

Reply:
The question is why do we need to separate states then?

Reply:
Diagnostics and transparency is one such reason.